ykcompile/src/lib.rs

```diff
@@ -108,6 +108,12 @@ impl TraceCompiler {
         }
     }
 
+    fn free_register(&mut self, local: &Local) {
+        if let Some(reg) = self.assigned_regs.get(local) {
+            self.available_regs.push(*reg);
```

Comment:
What about the hashmap storing register assignments? Does the entry there need to be freed too?

I guess it depends if a local index can be reused. I don't think they can...

Reply:
Yes, I noticed that only after I assigned you. Fix is already pushed.

Reply:
I'm suggesting that it may not be strictly necessary. Read again.

Reply:
They can be reused, e.g. by entering a function call (all arguments are assigned to $1..$N). So we should definitely clean up the `assigned_regs` too.

Though now that I think about it, entering a function call messes up `assigned_regs` already, i.e. we would have to reset and reload `assigned_regs` when entering/leaving a call, which we currently don't do. Well, investigating proper register allocation is next on my todo list anyways, so we should fix this properly.

Reply:
Agreed. The problem you point out above will go away with a proper allocator.